Add packages page

HaSistrunk opened the pull request:
Address #11 

Adds page for package details. This isn't totally complete, but I think a review at this stage would be useful. It's still missing data in the summary lists, because I'm not sure where to find it at the moment. Also there are no tests. 

One thing I did that we might want to change: I added some logic for badge color and text to be used at the top of the page to indicate the package's last known event outcome. We may want to move that logic somewhere else to be used by multiple pages/components. ALSO, there's an open question about what statuses we need.

I'm going to add a separate issue for updating the Table component to more easily add badge components to it for the outcome data.

HaSistrunk committed:
create package details page with table component

HaSistrunk committed:
add preliminary badge for outcome

HaSistrunk committed:
add summarylists to package page

HaSistrunk committed:
handle data that is undefined or null in summary lists

HaSistrunk committed:
improve card layout

helrond commented:
This looks pretty solid to me. I think you've identified the main issues.
- In terms of missing package data, the identifiers will be in an `identifiers` array. [This fixture](https://github.com/RockefellerArchiveCenter/digital_ingest_transformation/blob/base/tests/fixtures/updated/create_digital_object--digitization.json#L51-L59) should give you a sense of that. Note that keys are added as the package moves through the transformation process, rather than having a static set of keys with null values).
- The package file path is currently not recorded anywhere. Definitely something we can add, but I think there might be some challenges around in-progress packages, so I'm trying to remember the use case here.
- In terms of improving the `Table` component, I'm wondering what it would look like to just be able to pass a component or HTML directly in the data. That would mean you'd do the mapping of the initial data to whatever you wanted to pass to the Table outside of the Table component, but it might be better than baking in a lot of internal logic to that component.
- Was also looking at the Summary List - does it make sense to simplify this and have the items passed as an object, and then have the component take care of mapping those to SummaryListItems rather than having to use two components?

helrond commented:
Also I would definitely support moving that badge color setting somewhere else. Does it make sense to just put it in the badge component or do we need it in other places?

HaSistrunk committed:
update badge to handle event outcome text and color

helrond committed:
update summary list implementation

helrond committed:
linting

HaSistrunk committed:
add package external ids data

HaSistrunk commented:
Alright! We've now simplified the summaryList, moved the color logic into the Badge component (making the Badge specifically an OutcomeBadge, which is all we are using it for in this app) and added the external identifiers data to the package page. We determined that package file paths are not necessary at this time.

I'm actually now wondering if we even want the "status" outcome text to be in badges in the tables. It might actually be a more effective communication tool to just have the current outcome as THE badge on the page.

The only thing left, that I see, is to update the layout a bit since the two cards are uneven in terms of content. That might require taking the summaryLists out of cards in the component, but let me see...

HaSistrunk committed:
redesign package details page

HaSistrunk committed:
add badge text to clarify what it is

HaSistrunk commented:
Can I get some help writing tests for this page? I'm not sure the best way to handle mock data, which I presume we'll want to use for tests on other pages, too. I think we use jest.mock and provide mock response data somewhere?

helrond commented:
Yeah so I think we want to mock `fetchData` (rather than `axios` or `fetch`) because there are a lot of internals in that function that should probably be tested in isolation. I am not sure if the data will actually be reused in other places because the events page is going to be pulling from different endpoints, so I think it's okay to either hardcode the data returned directly using `mockReturn` or `mockImplementation`. One thing we will need to do (at least on this page) is work in some conditionals that look at the request URL and return the appropriate data, since we're fetching data from multiple endpoints.

The [PageRecord tests in DIMES](https://github.com/RockefellerArchiveCenter/dimes/blob/base/src/components/PageRecords/__tests__/PageRecords.test.js) might be a useful reference.

HaSistrunk committed:
update badge tests

HaSistrunk committed:
add className to SummaryList test

HaSistrunk committed:
add tests for package details page

HaSistrunk committed:
remove unused import

helrond reviewed (approved):
This looks good to me!
